From 61fc548fe1a323dd2344c8ae267e3ce05e86da7d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jakub=20Jank=C5=AF?= <janku.jakub.jj@gmail.com>
Date: Thu, 19 Oct 2017 20:33:16 +0200
Subject: [PATCH 38/55] Bump GLib version to 2.34

This is required for the following GMainLoop integration,
which utilizes some of the new functions/definitions, namely:

[definition][GLib version]
g_clear_pointer()   2.34
G_SOURCE_REMOVE     2.32
G_SOURCE_CONTINUE   2.32
g_unix_signal_add() 2.30

GLib version accross distributions:
- Fedora 26  2.52
- Fedora 25  2.50
- Fedora 24  2.48
- CentOS 7   2.46
- Debian 9   2.50

RHEL 6 is no longer supported,
so remove README.RHEL-5.

Acked-by: Frediano Ziglio <fziglio@redhat.com>
---
 Makefile.am   |  1 -
 README.RHEL-5 | 34 ----------------------------------
 configure.ac  |  2 +-
 3 files changed, 1 insertion(+), 36 deletions(-)
 delete mode 100644 README.RHEL-5

diff --git a/Makefile.am b/Makefile.am
index 45f7177..4689713 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -121,7 +121,6 @@ manpage_DATA =					\
 
 EXTRA_DIST =					\
 	NEWS					\
-	README.RHEL-5				\
 	data/70-spice-vdagentd.rules		\
 	data/spice-vdagent.desktop		\
 	data/spice-vdagentd			\
diff --git a/README.RHEL-5 b/README.RHEL-5
deleted file mode 100644
index 9b9ccc7..0000000
--- a/README.RHEL-5
+++ /dev/null
@@ -1,34 +0,0 @@
-spice-vdagent and RHEL-5 README
--------------------------------
-
-spice-vdagent can also be used inside RHEL-5 guests, if, and only if, they
-are hosted on a host which uses virtio-serial as the agent channel. Note
-that older hosts, such as qemu shipped with RHEL-5 for example, use
-a custom pci device (the spicevmc device) for the agent channel and
-there is no Linux support for this device so spice-vdagent does not work
-inside guests hosted on such hosts! Hosts based on RHEL-6 or Fedora 15
-and newer for example, are fine.
-
-spice-vdagent relies on ConsoleKit to determine the active session, since
-RHEL-5 has no ConsoleKit it cannot do this on RHEL-5, therefor only 1
-active X-session is supported with RHEL-5. If you try to start multiple
-sessions by running multiple X-servers the agent will disable itself.
-
-Since the RHEL-5 X-server does not support hotplugging of input devices,
-spice-vdagentd must be started before the X-server, so that it can create
-the uinput tablet device it uses for agent mouse mode.
-
-This also means that you must use a customized xorg.conf to teach the
-X-server about the uinput device. An example xorg.conf is included with
-spice-vdagent. After installing spice-vdagent, first start spice-vdagentd,
-and then copy this file to /etc/X11/xorg.conf and restart your X-server.
-
-Note: the sample xorg.conf assumes that the vm has been configured to *not*
-use a usb tablet device, since there is no need for one with the agent and
-usb emulation takes a significant amount of CPU.
-
-Building from source for RHEL-5
--------------------------------
-
-If you're building spice-vdagent from source on RHEL-5, you must pass
---disable-console-kit --enable-static-uinput to ./configure!
diff --git a/configure.ac b/configure.ac
index fbc20a9..d92b527 100644
--- a/configure.ac
+++ b/configure.ac
@@ -85,7 +85,7 @@ AC_ARG_ENABLE([static-uinput],
               [enable_static_uinput="$enableval"],
               [enable_static_uinput="no"])
 
-PKG_CHECK_MODULES([GLIB2], [glib-2.0 >= 2.28])
+PKG_CHECK_MODULES([GLIB2], [glib-2.0 >= 2.34])
 PKG_CHECK_MODULES(X, [xfixes xrandr >= 1.3 xinerama x11])
 PKG_CHECK_MODULES(SPICE, [spice-protocol >= 0.12.13])
 PKG_CHECK_MODULES(ALSA, [alsa >= 1.0.22])
-- 
2.16.1

